src/librustc_typeck/check/method/suggest.rs

```diff
@@ -282,6 +283,18 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                                               item_name,
                                               rcvr_expr,
                                               out_of_scope_traits);
+
+                if !lev_candidates.is_empty() {
+                    use rustc_data_structures::fx::FxHashSet;
+                    let mut dedup = FxHashSet();
```

Comment:
I think this `dedup` thing is redundant now...

Reply:
It should be - `candidate_method_names` performs deduplication.

Reply:
Removed.